### PR TITLE
feat: Added Option to delete Session/Sessions

### LINKED ISSUE #935: Option to delete Session
### Feature Request

**Feature Description**

User can long click a Session to delete it.

**Parent Issue**
#648 
   
**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #935 

Changes: 

1. Select item/items -> click on toolbar delete button -> click ok to confirm.
2. Long click on an item to select it, after it is selected then select subsequent items through click/longclick.
4. Click/Longclick on selected item to unselect it.
5. Database change listener reflects change in list when deleted.
6. Unit Tests added.

Screenshots for the change: 
20180515105627-7c011ad592 gif-2-mp4 com

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/data/session/Session.java [hunk 1]
```diff
@@ -75,4 +75,8 @@
     @JsonSerialize(using = ObservableStringSerializer.class)
     @JsonDeserialize(using = ObservableStringDeserializer.class)
     public ObservableString endsAt = new ObservableString();
+
+    @ColumnIgnore
+    @JsonIgnore
+    public ObservableBoolean selected = new ObservableBoolean();
```
Comment (srv-twry) [inline app/src/main/java/org/fossasia/openevent/app/data/session/Session.java]: Please use Boolean. We are removing the use of Observable fields.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/data/session/Session.java]: Also, move this property outside the model if possible

File: app/src/main/java/org/fossasia/openevent/app/core/session/list/SessionsPresenter.java [hunk 2]
```diff
@@ -67,12 +70,59 @@ private void listenChanges() {
...
+        }
+        else
+            deletionList.add(currentSession);
+
+        getView().changeToDeletingMode();
+    }
+
+    public void resetToDefaultState() {
+        for (int i = 0; i < deletionList.size(); i++)
+            unselectSession(deletionList.get(i));
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/session/list/SessionsPresenter.java]: Use for-each

File: app/src/main/java/org/fossasia/openevent/app/core/session/list/SessionsPresenter.java [hunk 3]
```diff
@@ -67,12 +70,59 @@ private void listenChanges() {
...
+            session.getSelected().set(false);
+    }
+
+    public void toolbarDeleteMode(Session currentSession) {
+        if (deletionList.contains(currentSession)) {
+            deletionList.remove(currentSession);
+            unselectSession(currentSession);
+        }
+        else
+            deletionList.add(currentSession);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/session/list/SessionsPresenter.java]: Use curly braces. Also move else to previous line

**Comment by Masquerade0097** [comment]
@iamareebjamal Please review.

**Comment by mishuvs** [comment]
@Masquerade0097 

> If an item is already selected then it can be deselected by long click.

Can I unselect by a simple click?

> If a user has selected some items and then it clicks anywhere in layout then the items gets deselected.

Accidental clicks can sometimes unselect all the items that the user has been clicking to delete.

**Comment by Masquerade0097** [comment]
@mishuvs 
> Can I unselect by a simple click?
Single click unselect all the selected items in the list. This is added intentionally. To unselect a selected item just long press on it again and it will be unselected.

**Comment by mishuvs** [comment]
Selecting and unselecting every item by long clicks seems counter-intuitive IMO. Only the first one should require a long click. What do you suggest @iamareebjamal?

**Comment by iamareebjamal** [comment]
I agree. The behavior should be consistent with standard apps

**Pull request merged**